File: LowLevelDesign/tic-tac-toe/src/strategy/ColWinningStrategy.ts
import { Board } from "../Board";
import { Player } from "../Player";
import { WinningStrategy } from "./WinningStrategy";

export class ColWinningStrategy  implements WinningStrategy {
    checkWinner(board: Board, player: Player): boolean {
        for(let col = 0; col < board.size; col++) {
            let colWin = true;
            for(let row = 0; row < board.size; row++) {
                if(board.board[row][col].getSymbol() !== player.symbol) {
                    colWin = false;
                    break;
                }
            }
            if(colWin) return true;
        }
        return false
    }
}

File: SystemDesignDiagrams/practice/LLD/tictactoe/src/main/java/com/tictactoe/strategies/ColWinningStrategy.java
package com.tictactoe.strategies;

import com.tictactoe.entities.Board;
import com.tictactoe.entities.Game;
import com.tictactoe.entities.Player;

public class ColWinningStrategy implements WinningStrategy {
    public boolean checkWinner(Game game) {
        Board board = game.getBoard();
        Player currentPlayer = game.getCurrentPlayer();
            for(int col = 0; col < board.getSize(); col++) {
                boolean isCol = true;
                for(int row= 0; row < board.getSize(); row++) {
                    if(board.getBoard()[row][col].getSymbol() != currentPlayer.getSymbol()) {
                        isCol = false;
                        break;
                    }
                }
                if(isCol) {
                    return true;
                }
            }


        return false;
    }
}
